Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1721,6 +1727,22 @@ private extension Purchases {
         }
 
         self.updateOfferingsCache(isAppBackgrounded: isAppBackgrounded)
+
+        self.fetchAndCacheAppTransactionIfNeeded()
+    }
+
+    func fetchAndCacheAppTransactionIfNeeded() {
+        if self.systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS,
+           self.systemInfo.originalAppVersion == nil,
+           self.systemInfo.originalAppPurchaseDate == nil {
+            if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+                _ = Task<Void, Never> {
+                    let appTransaction = await self.transactionFetcher.appTransaction
+                    systemInfo.originalAppVersion = appTransaction?.originalApplicationVersion
+                    systemInfo.originalAppPurchaseDate = appTransaction?.originalPurchaseDate
+                }
+            }
+        }
```

Comment:
AppTransaction will be fetched on first launch (if not backgrounded) and on each subsequent foregrounding, if it's not already cached.